### PR TITLE
Restore Legacy Custom Layout option

### PR DESCRIPTION
**Pull request by rtiangha**
Apparently [a lot of people](https://www.reddit.com/r/Citra/comments/afsq4a/various_custom_screen_layouts/) took advantage of the hidden Custom Layout functionality and took the time to do the math to figure out the optimum placement for various resolutions of displays. So if they were to switch to or upgrade to the current version of Lime3DS, they would find that their profiles no longer worked.

So in order to maintain backwards compatibility with those who have already specified Custom Layout profiles manually, this PR brings back the logic to handle the legacy Custom Layout functionality, similar to how the original project chose to support both the old way and new way of dealing with custom texture hashes.

The only meaningful change is that it brings back the old variable names and moves processing of the new variable names to a function called `NewCustomFrameLayout`. So if a user has a `.ini` file that's already set up using the old method, that will continue to work with no intervention. Conversely, if someone used the UI to set up the new method and never used the old method to begin with, that will continue to work. All of this should be completely transparent to the user.

In the case that both Legacy Custom Layout and New Custom Layout options have been set, the Legacy Custom Layout options will take priority and in that case, if the user wishes to migrate to the new method, they'll need to clear out the old variable values in their `.ini` file first.

This PR is really just to reduce the support burden of people chiming in after updating wondering why their Custom Layout profiles no longer work.

### CODE REVIEW

**Comment by YGOmod** [comment]
can we have something like citra mmj (custom layouts for both landscape and portrait)

currently on android when you use custom layout and switch from one (portrait or landscape) mode to another, layout doesn't change.. maybe just separating them as portrait_top_left and landscape_top_left would make things work as intended.. I'm sending screenshot of current state

Landscape
Screenshot_20240531_224412_Lime3DS

Portrait
Screenshot_20240531_224400_Lime3DS

**Comment by OpenSauce04** [comment]
I'm not personally comfortable with adding this legacy code. It makes the code for custom layouts extremely confusing and unreadable with regard to the many variable names being thrown around, as it is not immediately apparent which layout system is being referred to at any given time unless you have pre-existing knowledge of the variable names and this clash between implementations.

In order to avoid a mess, I believe that it is for the best that the old system be dropped in favour of the new system. An argument can be made that this may impact usage between different Citra forks, however I believe that in order to make innovations with the emulator, we can't be held back by historical decisions that Citra has made with the codebase or cater to other projects which are run by different people and have different goals.

In other words, I want to prioritize only a single implementation/integration of any given feature rather than simultaneously maintaining older implementations in a way which fragments the user experience.

It's not that hard to convert the values anyway: You simply have to subtract the Left/Top value from the Right/Bottom value and it will work in the new system.

Please feel free to make a case as to why I should believe otherwise.

**Comment by rtiangha** [comment]
Your call. I don't feel strongly either way; just wanted to pre-emptively reduce the support burden and it's not like this code would change much in the future, even if it is a duplication.

So instead, I feel like there should be some permanent documentation in an FAQ or similar that walks users on how to migrate their profiles. I don't think it's fair that a breaking change like this should be introduced without warning or a way relayed to deal with it. Perhaps it'd be fair if Lime3DS never supported the legacy way to begin with when people switched, but it did for a few versions. Then again, who really knows how many users would actually be affected by this.
